File: N_NumbersDiff_E_O/Diff.py
"""
    Author : Pakshal Shashikant Jain 
    Date : 07/06/2023
    Program :  Accept N numbers from user and return difference between summation
               of even elements and summation of odd elements.
               Input : N : 6
               Elements : 85 66 3 80 93 88
               Output : 53 (234 - 181) 
"""

################################################################################################################################################

#Class Declaration
class EvenOdd :
    isize = 0;
    arr = [];

    #Constructor of Class
    def __init__(self) :
        self.isize = 0;
        self.arr = [0];
    
    #Behaviour of Class To Take Value From User
    def accept(self) :
        i = 0;
        No = 0;

        #Taking Input From User
        print("Enter Number of Elements You Want in Array : ");
        self.isize = int(input());

        #Inserting Value in Array 
        print("Enter Elements in array");
        for i in range(self.isize) :
            print("Enter Element Number : %d"%(i + 1));
            No = int(input());

            self.arr.append(No);
        
    #Behaviour of Classs To Calculate Difference 
    def Diff(self) :
        i = 0;
        diff = 0;
        SumEven = 0;
        SumOdd = 0;

        #Logic To Calculate Difference 
        for i in range(self.isize + 1) :
            if((self.arr[i] % 2) == 0) :
                SumEven = SumEven + self.arr[i];
            else :
                SumOdd = SumOdd + self.arr[i];
        
        diff = SumEven - SumOdd;

        return diff;

######################################################################################################################################################

#Main Function
def main() :
    iret = 0;

    print("Jay Ganesh.....");

    #Creating Object of Even Odd Class
    eobj = EvenOdd();

    #Calls To Behaviours of Class
    eobj.accept();
    iret = eobj.Diff();

    #Printing Difference Calculated
    print("Difference Between Even and Odd Elements is : %d"%iret);

#####################################################################################################################################################################

#Call Back Method To Call Main Function
if __name__ == "__main__" :
    #Call To Main Function
    main();

File: N_NumbersDiff_E_O/Diff.java
/*
    Author : Pakshal Shashikant Jain 
    Date : 07/06/2023
    Program :  Accept N numbers from user and return difference between summation
               of even elements and summation of odd elements.
               Input : N : 6
               Elements : 85 66 3 80 93 88
               Output : 53 (234 - 181) 
*/

//Required Libraries
import java.lang.*;
import java.util.*;

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class EvenOdd
{
    //Characteristics of Class
    private int isize;
    private int arr[];   

    //Constructor of Class
    EvenOdd()
    {
        this.isize = 0;
    }

    //Behaviour of Class To Take input From From User
    void Accept()
    {
        int i = 0;

        //Creating object of Scanner Class To Take Input From User
        Scanner sobj = new Scanner(System.in);

        //Taking Input From User
        System.out.println("Enter Number of Elements You Want in Array");
        isize = sobj.nextInt();

        this.arr = new int[this.isize];

        System.out.println("Enter ELements in Array....");
        for(i = 0;i < this.isize;i++)
        {
            System.out.printf("Enter Element Number %d \n",i + 1);
            this.arr[i] = sobj.nextInt();
        }
    }

   //Behaviour of Class To Calcualte Difference 
    int Diff()
    {
        int i = 0;
        int SumEven = 0;    
        int SumOdd = 0;
        int diff = 0;

        //Logic To Calculate Difference
        for(i = 0;i < isize;i++)
        {
            if((this.arr[i] % 2) == 0)
            {   
                SumEven = SumEven + this.arr[i];
            }
            else 
            {
                SumOdd = SumOdd + this.arr[i];
            }
        }

        diff = SumEven - SumOdd;

        return diff;
    }
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Class
class Diff
{
    //Main Function
    public static void main(String args[])
    {
        int iret = 0;

        System.out.println("Jay Ganesh....");

        //Creating object of EvenOdd Class
        EvenOdd eobj = new EvenOdd();

        //Call To Behaviours if Class
        eobj.Accept();
        iret = eobj.Diff();

        //Printing Difference Calculated
        System.out.printf("Difference Between Smmation of Even and Odd Elements is : %d",iret);

        //Deallcating Memory Of Object
        eobj = null;
    }
}